### ISSUE TITLE
Using ERA5 DJFM -> Error in repcheck[, 3] : subscript out of bounds

### ISSUE THREAD

**Issue by JanStreffing** [oloapinivad/MiLES#6]
I'm trying to compare blocking indices to a none standard climatology (DJFM ERA5 1979-2019) and came across the following issue:

> ###########################################################
MiLES is running on:
ERA5   1979 2019
Active options are:
block figures doforceanl
###########################################################
Full filename is: /p/largedata/hhb19/jstreffi/runtime/oifsamip/MILES/data/zg50000//ERA5/zg50000_ERA5_fullfile.nc
50000
zg50000 NetCDF data seems there, avoid universal_prepare.sh
real	0m0.002s
user	0m0.002s
sys	0m0.000s
Working on DJFM season
List of 6
 $ field    : num [1:288, 1:144, 1:4971] 51416 51415 51414 51413 51413 ...
 $ lon      : num [1:288] -179 -178 -176 -175 -174 ...
 $ lat      : num [1:144(1d)] -89.4 -88.1 -86.9 -85.6 -84.4 ...
 $ plev     : num [1(1d)] 50000
 $ time     : 'PCICt' num [1:4971(1d)] 1979-01-01 09:00:00 1979-01-02 09:00:00 1979-01-03 09:00:00 1979-01-04 09:00:00 ...
  ..- attr(*, "cal")= chr "proleptic_gregorian"
  ..- attr(*, "tzone")= chr "GMT"
  ..- attr(*, "units")= chr "secs"
 $ var_units: chr "m**2 s**-2"
NULL
[1] "Time Array Built"
[1] "Length: 4971"
[1] "From 1979-01-01 09:00:00 to 2019-12-31 09:00:00"
[1] "xcritical: 2.5"
[1] "xreso: 1.25"
[1] "ycritical: 2.5"
[1] "yreso: 1.25"
[1] "Tibaldi and Molteni (1990) index..."
[1] "D'Andrea et al. (1998) index..."
[1] "--------------------------------------------------"
[1] "Blocking indices..."
[1] "gradients"
[1] "anomalies and threshold..."
[1] "Computing D12 instantaneous index ..."
[1] "Computing ExtraD12 instantaneous index ..."
[1] "Computing S04 instantaneous index ..."
[1] "Diagnostics..."
[1] "Creating abind dataset ..."
[1] "Computing Rossby Wave Breaking..."
[1] "dim(repcheck): 4078745" "dim(repcheck): 2"      
[1] "dim(ii): "
[1] "steprwb: 6"
Error in repcheck[, 3] : subscript out of bounds
Calls: miles.block.multiple -> cbind
Execution halted

This seems to stem from https://github.com/oloapinivad/MiLES/blob/887c128c9c68f44b7f832a25d51abcd9297feebd/script/block_multiple_fast.R#L209 It would appear that repcheck is missing one of it's dimension but it's not clear to me where the root cause of the problem lies. I somewhat suspect that it may be with the input file I provide:

> cdo sinfo zg50000_ERA5_fullfile.nc
   File format : NetCDF2
    -1 : Institut Source   T Steptype Levels Num    Points Num Dtype : Parameter ID
     1 : ECMWF    unknown  v instant       1   1     41472   1  F32  : 129.128       
   Grid coordinates :
     1 : lonlat                   : points=41472 (288x144)
                              lon : 0 to 358.75 by 1.25 degrees_east  circular
                              lat : -89.375 to 89.375 by 1.25 degrees_north
   Vertical coordinates :
     1 : pressure                 : levels=1
                             plev : 50000 Pa
   Time coordinate :  14975 steps
     RefTime =  1979-01-01 00:00:00  Units = hours  Calendar = proleptic_gregorian  Bounds = true
  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss  YYYY-MM-DD hh:mm:ss
  1979-01-01 09:00:00  1979-01-02 09:00:00  1979-01-03 09:00:00  1979-01-04 09:00:00
  1979-01-05 09:00:00  1979-01-06 09:00:00  1979-01-07 09:00:00  1979-01-08 09:00:00
  1979-01-09 09:00:00  1979-01-10 09:00:00  1979-01-11 09:00:00  1979-01-12 09:00:00
  1979-01-13 09:00:00  1979-01-14 09:00:00  1979-01-15 09:00:00  1979-01-16 09:00:00
  1979-01-17 09:00:00  1979-01-18 09:00:00  1979-01-19 09:00:00  1979-01-20 09:00:00
  1979-01-21 09:00:00  1979-01-22 09:00:00  1979-01-23 09:00:00  1979-01-24 09:00:00
  1979-01-25 09:00:00  1979-01-26 09:00:00  1979-01-27 09:00:00  1979-01-28 09:00:00
  1979-01-29 09:00:00  1979-01-30 09:00:00  1979-01-31 09:00:00  1979-02-01 09:00:00
  1979-02-02 09:00:00  1979-02-03 09:00:00  1979-02-04 09:00:00  1979-02-05 09:00:00
  1979-02-06 09:00:00  1979-02-07 09:00:00  1979-02-08 09:00:00  1979-02-09 09:00:00
  1979-02-10 09:00:00  1979-02-11 09:00:00  1979-02-12 09:00:00  1979-02-13 09:00:00
  1979-02-14 09:00:00  1979-02-15 09:00:00  1979-02-16 09:00:00  1979-02-17 09:00:00
  1979-02-18 09:00:00  1979-02-19 09:00:00  1979-02-20 09:00:00  1979-02-21 09:00:00
  1979-02-22 09:00:00  1979-02-23 09:00:00  1979-02-24 09:00:00  1979-02-25 09:00:00
  1979-02-26 09:00:00  1979-02-27 09:00:00  1979-02-28 09:00:00  1979-03-01 09:00:00
   ................................................................................
   ................................................................................
   ................................................................................
   ......
  2019-11-03 09:00:00  2019-11-04 09:00:00  2019-11-05 09:00:00  2019-11-06 09:00:00
  2019-11-07 09:00:00  2019-11-08 09:00:00  2019-11-09 09:00:00  2019-11-10 09:00:00
  2019-11-11 09:00:00  2019-11-12 09:00:00  2019-11-13 09:00:00  2019-11-14 09:00:00
  2019-11-15 09:00:00  2019-11-16 09:00:00  2019-11-17 09:00:00  2019-11-18 09:00:00
  2019-11-19 09:00:00  2019-11-20 09:00:00  2019-11-21 09:00:00  2019-11-22 09:00:00
  2019-11-23 09:00:00  2019-11-24 09:00:00  2019-11-25 09:00:00  2019-11-26 09:00:00
  2019-11-27 09:00:00  2019-11-28 09:00:00  2019-11-29 09:00:00  2019-11-30 09:00:00
  2019-12-01 09:00:00  2019-12-02 09:00:00  2019-12-03 09:00:00  2019-12-04 09:00:00
  2019-12-05 09:00:00  2019-12-06 09:00:00  2019-12-07 09:00:00  2019-12-08 09:00:00
  2019-12-09 09:00:00  2019-12-10 09:00:00  2019-12-11 09:00:00  2019-12-12 09:00:00
  2019-12-13 09:00:00  2019-12-14 09:00:00  2019-12-15 09:00:00  2019-12-16 09:00:00
  2019-12-17 09:00:00  2019-12-18 09:00:00  2019-12-19 09:00:00  2019-12-20 09:00:00
  2019-12-21 09:00:00  2019-12-22 09:00:00  2019-12-23 09:00:00  2019-12-24 09:00:00
  2019-12-25 09:00:00  2019-12-26 09:00:00  2019-12-27 09:00:00  2019-12-28 09:00:00
  2019-12-29 09:00:00  2019-12-30 09:00:00  2019-12-31 09:00:00
cdo sinfo: Processed 1 variable over 14975 timesteps [0.59s 40MB]

Do you see anything obviously wrong with what I've done here?

Best regards, 
Jan

**Comment by oloapinivad** [comment]
Hi @JanStreffing 

I need to investigate it a bit more, but I guess this comes from the fact that your are testing the 1.25 grid instead of the standard 2.5. I made a few modification to make the code work also with the 1.25, but I never tested it and I guess you spotted one issue. I will give it a look during the week, let me know if you can solve it running with the 2.5 grid. 

Cheers,
Paolo

**Comment by oloapinivad** [comment]
Hi @JanStreffing 

I made a test with ERA5 at 1.25 resolution, and everything worked smoothly so I cannot reproduce your issue. Looking at your log, I noticed that there is an inconsistency in the initial file, since it goes from -89 to 89, while the files processed by "assimilate.sh" should work only in the Northern Hemisphere. Also it is shown as geopotential, not geopotential height (and this will won't work in the gradient computations). Did you pre-process the file with the wrapper?

**Comment by JanStreffing** [comment]
Hi @oloapinivad 

Thank you for taking the time and investigating. I just ran a remap to 144x73 on the ERA5 data and that worked. 144x72 did not. Since the 1.25° data was 288x144 I assume this regular number of latitudes presents the problem causing the issue I observed. I presume you used 288x145 for your test? 

> real	35m11.902s
> user	0m41.791s
> sys	0m4.378s
> ###########################################################
>  MiLES is over, go and enjoy your life outside! 
> ###########################################################

For completeness sake: I do use the wrapper scripts and it makes a selection to the Northern Hemisphere only as required.

Cheers, Jan
